shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -137,7 +141,15 @@ - (instancetype)initWithNibName:(NSString*)nibNameOrNil bundle:(NSBundle*)nibBun
 }
 
 - (instancetype)initWithCoder:(NSCoder*)aDecoder {
-  return [self initWithProject:nil nibName:nil bundle:nil];
+  self = [super initWithCoder:aDecoder];
```

Comment:
The properties should be pulled off `aDecoder` at initialization (with copies where appropriate)
```obj-c
  _viewOpaque = aDecoder->_viewOpaque;
  _weakFactory = aDecoder->_weakFactory;
  _engine = aDecoder->_engine;
  _flutterView = aDecoder->_flutterView;
  _isDisplayingFlutterUI = [aDecoder decodeObjectForKey:@"isDisplayingFlutterUI"];
  _isHomeIndicatorHidden = [aDecoder decodeObjectForKey:@"isHomeIndicatorHidden"];
  _isPresentingViewControllerAnimating = [aDecoder decodeObjectForKey:@"isPresentingViewControllerAnimating"];
```
not re-set up as if it's a newly initialized object (it's being deserialized).

Reply:
The NSCoder doesn't have a representation of a full object.  It just has what is coming from the xib, so it won't have things like a _weakFactory.  The default initWithCoder will use key-value to set properties that have been specified in the xib, you don't need to create custom readers for them.

Reply:
I never did grok `UIViewController` or `NSViewController`.  Looked at more docs, you're right.